org.eclipse.lsp4e.test/src/org/eclipse/lsp4e/test/references/FindReferencesTest.java

```diff
@@ -111,7 +108,6 @@ public void findReferencesNonBlocking() throws Exception {
 		uiThreadActiveChecker.start();
 		try {
 			assertNotNull("Search Result view not found", findSearchResultView(5000));
-			assertTrue("UI Thread was frozen", runCount.intValue() > 1000);
```

Comment:
I have checked the new code and I think the reason why I need to set this to a lower value is that the new code blocks the UI less than the previous one, and findSearchResultView returns faster than before, thus the uiThreadActiveChecker does not have time to increase the time high enough.

I have put some breakpoints to make sure of that, and I am quite sure that creating the lambda in the body of execturor.collectAll in LSSearchQuery.run is not being run in the UI Thread. Also we run the query in the background with NewSearchUI.runQueryInBackground(query).

Do you agree that the test is not actually measuring frozen UI, but rather the number of iterations in the while loop in the uiThreadActiveChecker, and that being small is not the same as a frozen UI?